### PR TITLE
Add in/notIn for subqueries

### PR DESCRIPTION
**Pull request by timowest**

### CODE REVIEW

**Comment by Shredder121** [comment]
I think adding at least a compile snippet is good.

**Comment by timowest** [comment]
This doesn't need to be backported, since in Querydsl 3 subqueries are collection expressions.

File: querydsl-core/src/test/java/com/querydsl/core/types/ExpressionUtilsTest.java [hunk 1]
```diff
@@ -148,6 +150,12 @@ public void In() {
     }
 
     @Test
+    public void In_SubQuery() {
+        String s = ExpressionUtils.in(str, new SubQueryExpressionImpl<String>(String.class, new DefaultQueryMetadata())).toString();
+        assertTrue(s.startsWith("str in com.querydsl.core.DefaultQueryMetadata@c"));
```
Comment (Shredder121) [inline querydsl-core/src/test/java/com/querydsl/core/types/ExpressionUtilsTest.java]: This is a strange toString, did you intend to make it that way?

Comment (timowest) [inline querydsl-core/src/test/java/com/querydsl/core/types/ExpressionUtilsTest.java]: Alternatively we can use some SQL-like syntax to serialize module independent subquery expressions.

Now it just calls just toString() on the metadata.

Comment (timowest) [inline querydsl-core/src/test/java/com/querydsl/core/types/ExpressionUtilsTest.java]: But maybe that change could be done in a different PR.

**Comment by Shredder121** [comment]
Well, to me it looks good so far.
@johnktims what's your view?

**Comment by johnktims** [comment]
Looks good to me. I'll merge it.

**Pull request merged**